### PR TITLE
Metadata

### PR DESCRIPTION
**Pull request by pjhartzell**
**Description:**
Add metadata and extensions.

**PR checklist:**

- [x] Code is formatted (run `scripts/format`).
- [x] Code lints properly (run `scripts/lint`).
- [x] Tests pass (run `scripts/test`).
- [ ] Documentation has been updated to reflect changes, if applicable.
- [ ] Changes are added to the [CHANGELOG](../CHANGELOG.md).

### CODE REVIEW

File: src/stactools/nclimgrid/daily_stac.py [hunk 1]
```diff
@@ -195,13 +195,26 @@ def daily_base_item(year: int, month: int, day: int, status: Status) -> Item:
...
     item_id = f"{year}{month:02d}-grd-{status.value}-{day:02d}"
-    item_time = datetime(year, month, day, tzinfo=timezone.utc)
-    item = Item(id=item_id,
-                properties={},
-                geometry=WGS84_GEOMETRY,
-                bbox=WGS84_BBOX,
-                datetime=item_time,
-                stac_extensions=[])
+    item_start_time = datetime(year, month, day, tzinfo=timezone.utc)
+    item_end_time = datetime(year, month, day, 23, 59, 59)
```
Comment (gadomski) [inline src/stactools/nclimgrid/daily_stac.py]: Probably worth specifying the time zone for the end time as well.

File: src/stactools/nclimgrid/daily_stac.py [hunk 2]
```diff
@@ -195,13 +195,26 @@ def daily_base_item(year: int, month: int, day: int, status: Status) -> Item:
...
+
+    item = Item(
+        id=item_id,
+        properties={},
+        geometry=constants.WGS84_GEOMETRY,
+        bbox=constants.WGS84_BBOX,
+        datetime=item_start_time,  # start of day is nominal
```
Comment (gadomski) [inline src/stactools/nclimgrid/daily_stac.py]: @matthewhanson can you remind me on the current best practice for time-ranged datasets w/o a single "nominal" time? I remember some discussion about including a `datetime` anyways just because so many tools rely on `datetime` (instead of checking inside `start_datetime` and `end_datetime`).

**Review by gadomski — CHANGES_REQUESTED** [review]

File: src/stactools/nclimgrid/daily_stac.py (hunk 1 above)
Comment (pjhartzell) [reply src/stactools/nclimgrid/daily_stac.py]: Oops. Yeah. Fixed.

File: src/stactools/nclimgrid/daily_stac.py (hunk 2 above)
Comment (pjhartzell) [reply src/stactools/nclimgrid/daily_stac.py]: Yeah, I wasn't super confident on what to use for the `datetime`. Figured most people would search by start of month or start of day when searching for monthly or daily data. But perhaps the middle of the month or day is better. Decision to include `datetime` as well as the `start_datetime` and `end_datetime` based on the STAC Best Practices doc: https://github.com/radiantearth/stac-spec/blob/master/best-practices.md#datetime-selection

> The datetime field in a STAC Item's properties is one of the most important parts of a STAC Item, providing the T (temporal) of STAC. And it can also be one of the most confusing, especially for data that covers a range of times. For many types of data it is straightforward - it is the capture or acquisition time. But often data is processed from a range of captures - drones usually gather a set of images over an hour and put them into a single image, mosaics combine data from several months, and data cubes represent slices of data over a range of time. For all these cases the recommended path is to use start_datetime and end_datetime fields from [common metadata](https://github.com/radiantearth/stac-spec/blob/master/item-spec/common-metadata.md#date-and-time-range). The specification does allow one to set the datetime field to null, but it is strongly recommended to populate the single datetime field, as that is what many clients will search on. If it is at all possible to pick a nominal or representative datetime then that should be used. But sometimes that is not possible, like a data cube that covers a time range from 1900 to 2000. Setting the datetime as 1950 would lead to it not being found if you searched 1990 to 2000.

Comment (matthewhanson) [reply src/stactools/nclimgrid/daily_stac.py]: The spec says that if there's no single datetime that makes sense for data that covers a temporal range to set `datetime` to null, in which case `start_datetime` and `end_datetime` are required.

I use the term "nominal" datetime, which may not be defined in the spec, to refer to a case where even though it's a range of data it can still be represented by a single datetime.  Modis is one example with it's weighted temporal windows.

Basically it comes down to "would someone query a single date and expect Items from this dataset to be returned that represent that single date?"

Comment (pjhartzell) [reply src/stactools/nclimgrid/daily_stac.py]: > Basically it comes down to "would someone query a single date and expect Items from this dataset to be returned that represent that single date?"

Daily data, yes => Remove the `start_datetime` and `end_datetime`, leave the `datetime`.
Monthly data, no => Set `datetime` to null, leave the `start_datetime` and `end_datetime`.

File: src/stactools/nclimgrid/monthly_stac.py [hunk 3]
```diff
@@ -140,16 +141,39 @@ def monthly_base_item(year: int, month: int) -> Item:
...
+                    "start_datetime": item_start_datetime,
+                    "end_datetime": item_end_datetime,
+                },
+                geometry=constants.WGS84_GEOMETRY,
+                bbox=constants.WGS84_BBOX,
+                datetime=None,
                 stac_extensions=[])
+
+    # item.common_metadata.start_datetime = item_start_time
+    # item.common_metadata.end_datetime = item_end_time
```
Comment (gadomski) [inline src/stactools/nclimgrid/monthly_stac.py]: Should these be uncommented?

Comment (pjhartzell) [reply src/stactools/nclimgrid/monthly_stac.py]: Nope. Deleted. Done.

**Review by gadomski — APPROVED** [review]
